test/cbmc/stubs/core_mqtt_stubs.c

```diff
@@ -0,0 +1,149 @@
+            break;
+
+        default:
+            break;
+    }
+
+    return isValid;
+}
+
+MQTTStatus_t MQTT_Init( MQTTContext_t * pContext,
+                        const TransportInterface_t * pTransportInterface,
+                        MQTTGetCurrentTimeFunc_t getTimeFunction,
+                        MQTTEventCallback_t userCallback,
+                        const MQTTFixedBuffer_t * pNetworkBuffer )
+{
+    MQTTStatus_t status;
+
+    __CPROVER_assert( pContext != NULL,
+                      "MQTT Context is not NULL." );
+
+    ( void ) memset( pContext, 0x00, sizeof( MQTTContext_t ) );
+
+    pContext->connectStatus = MQTTNotConnected;
+    pContext->transportInterface = *pTransportInterface;
+    pContext->getTime = getTimeFunction;
+    pContext->appCallback = userCallback;
+    pContext->networkBuffer = *pNetworkBuffer;
+
+    /* Zero is not a valid packet ID per MQTT spec. Start from 1. */
+    pContext->nextPacketId = 1;
+
+    return status;
+}
+
+MQTTStatus_t MQTT_ProcessLoop( MQTTContext_t * pContext,
+                               uint32_t timeoutMs )
+{
+    MQTTStatus_t status;
+    MQTTPacketInfo_t * pPacketInfo;
+    MQTTDeserializedInfo_t * pDeserializedInfo;
+    size_t remainingLength;
+    MQTTPublishInfo_t * pPublishInfo;
+    uint16_t topicNameLength;
+    size_t payloadLength;
+    static bool terminate = false;
+
+    __CPROVER_assert( pContext != NULL,
+                      "MQTT Context is not NULL." );
+
+    /* Only one packet per MQTT_ProcessLoop is received for the proof and it will be enough
+     * to prove the memory safety. The second invocation of MQTT_ProcessLoopreturns without
+     * invoking the appCallback. */
+    if( terminate == false )
+    {
+        pPacketInfo = malloc( sizeof( MQTTPacketInfo_t ) );
+        __CPROVER_assume( pPacketInfo != NULL );
+        __CPROVER_assume( isValidIncomingMqttPacket( pPacketInfo->type ) );
+        __CPROVER_assume( remainingLength < 64U );
+
+        /* SUBACK codes will be after 2 bytes. */
+        if( pPacketInfo->type == MQTT_PACKET_TYPE_SUBACK )
+        {
+            __CPROVER_assume( remainingLength > 2U );
+        }
+
+        pPacketInfo->pRemainingData = malloc( remainingLength );
+        __CPROVER_assume( pPacketInfo->pRemainingData != NULL );
+        pPacketInfo->remainingLength = remainingLength;
+
+        pDeserializedInfo = malloc( sizeof( MQTTDeserializedInfo_t ) );
+        __CPROVER_assume( pDeserializedInfo != NULL );
+
+        if( pPacketInfo->type == MQTT_PACKET_TYPE_PUBLISH )
+        {
+            pPublishInfo = malloc( sizeof( MQTTPublishInfo_t ) );
+            __CPROVER_assume( pPublishInfo != NULL );
+
+            __CPROVER_assume( topicNameLength < 32U );
+            pPublishInfo->pTopicName = malloc( topicNameLength );
+            __CPROVER_assume( pPublishInfo->pTopicName != NULL );
+            pPublishInfo->topicNameLength = topicNameLength;
+
+            __CPROVER_assume( payloadLength < 64U );
```

Comment:
Also for the magic numbers

Reply:
Yes That is a good thought. I will update the PR.